src/assets/githubAction/githubActionBeginner.md

```diff
@@ -0,0 +1,125 @@
+---
+title: Github Actions - CI & CD
+authorName: Nabin Ale
+authorAvatar : https://avatars.githubusercontent.com/u/61624650?s=400&u=8f992a9ca189af19c40cb070a5af1e02669d3884&v=4
+authorLink: https://github.com/nabim777
+createdAt: Oct 30, 2023
+tags: CI, CD, githubAction
+banner: https://raw.githubusercontent.com/JankariTech/blog/master/src/imgs/fallback_banner.png
+---
+GitHub Actions is an automation and CI/CD service provided by GitHub. It allows us to automate our software development process or workflows, from building and testing code to deploying it to different environments. It's a really effective tool that can both increase software quality and save us time.
+
+## Features in github actions:
+- It allows you to automate your Software Development Life Cycle (SDLC) processes or workflows.
+- It simplifies CI/CD integration, enabling seamless DevOps practices.
+
+## Components of github actions:
+1. Workflows
+2. Events
+3. Jobs
+4. Steps
+5. Actions
+6. Runners
+
+![components](/src/assets/githubAction/images/components_of_github_action.png)
+
+
+### 1. Workflows:
+Workflows are automated procedures that you add to your repository. Workflows will run when triggered by an event in your repository, or they can be triggered manually, or at a defined schedule. It can be used to build, test, package, release, or deploy a project on Github.
+
+### 2. Events:
+Events are specific activities like code pushes, pull requests creation, issue creation, issue comments, etc. that triggers a workflow.
+
+### 3. Jobs
+Jobs are the set of steps that were executed in same runner. A workflow with multiple jobs will run in parallel which is default, but it can be configured so that jobs will run sequentially.
+
+
+### 4. Steps
+A step is an individual task that can run the commands in a job. These can be actions or commands.
+
+### 5. Actions
+Actions are the standalone commands for GitHub Actions platform that perform a complex but frequently repeated task. Actions helps to reduce the repetitive code in workflows.
+
+### 6. Runner
+Runner is a github actions server where our workflows run. A runner listens for available jobs. Each runner runs jobs at a time, and reports the progress, logs, and results back to github. Github hosted runners are based on Ubuntu, Linux, Microsoft Windows, and macos.
```

Comment:
```suggestion
A runner is a GitHub actions server where our workflows run. A runner listens for available jobs. Each runner runs jobs at a time, and reports the progress, logs, and results back to GitHub. GitHub hosted runners are based on Ubuntu, Linux, Microsoft Windows, and Mac OS.
```

"Each runner runs jobs at a time" - what do you want to say - "1 job at a time", "multiple jobs at a time"?

Reply:
I meant to say "multiple jobs at a time" :sweat_smile: